activesupport/lib/active_support/cache.rb

```diff
@@ -647,7 +647,7 @@ def expanded_key(key)
             if key.size > 1
               key = key.collect { |element| expanded_key(element) }
             else
-              key = key.first
+              key = expanded_key(key.first)
```

Comment:
Not sure if it's worth to keep this optimization - we're saving on `#collect`, but paying the price of calling `#size` and `#first`. How about simply
```ruby
case key
  when Array
    key = key.map { |element| expanded_key(element) }
  when Hash
    # ...
```

Reply:
If you’re going to suggest a perf optimization or deoptimization please provide at least a microbenchmark and the results. I recommend using “benchmark ips”.

In this case the “slow” version allocates memory which is usually drastically slower than any calculations that use CPU time. I encourage you to write a benchmark for yourself though and not take my word.

I have another PR that standardized both cases and eliminates the allocation https://github.com/rails/rails/pull/33944

Reply:
You're right @schneems 👍